### PR TITLE
[TT-6702] Add hidden version base id

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4383/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3376983628)

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -407,6 +407,7 @@ type VersionDefinition struct {
 	StripPath           bool              `bson:"strip_path" json:"strip_path"` // Deprecated. Use StripVersioningData instead.
 	StripVersioningData bool              `bson:"strip_versioning_data" json:"strip_versioning_data"`
 	Versions            map[string]string `bson:"versions" json:"versions"`
+	BaseID              string            `bson:"base_id" json:"-"`
```
Comment (titpetric) [inline apidef/api_definitions.go:410]: bson doesn't match json (needs to be `-`, or json needs an update)?

**Review by jeffy-mathew — APPROVED** [review]

File: apidef/api_definitions.go (hunk 1 above)
Comment (furkansenharputlu) [reply apidef/api_definitions.go:410]: It was by purpose of that we don't want to show base api id to user in API definition.

Comment (titpetric) [reply apidef/api_definitions.go:410]: Add the intention in a comment please to make it clear it's intent 👍

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4383/merge
Commit: 751a7fc17b5c13a3500a937a13dc7570575f37fb 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3385538889)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4383/merge
Commit: 2a9a9fbf39800b62aa197790d5c5b829d9088679 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3385590688)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: f86b66b047ec0d7add5bedd4c24ac03f4d65d6c7 Add hidden version base id (#4383)
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3385680070)

**Pull request merged**